src/core/qgsgpkgflusher.cpp

```diff
@@ -61,7 +61,25 @@ void QgsGpkgFlusher::onLayersAdded( const QList<QgsMapLayer *> layers )
     if ( vl && vl->dataProvider() )
     {
       QString dataSourceUri = vl->dataProvider()->dataSourceUri();
-      QString filePath = dataSourceUri.left( dataSourceUri.indexOf( '|' ) );
+
+      QString filePath;
+      if( dataSourceUri.contains( QStringLiteral(".sqlite"), Qt::CaseInsensitive) )
```

Comment:
Should this be `endsWith`?

Reply:
It does not not end with `.sqlite`, there are table data as well.